codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/CborParserGenerator.kt

```diff
@@ -619,7 +653,18 @@ class CborParserGenerator(
                     if (returnSymbolToParse.isUnconstrained) {
                         rust("Ok(builder)")
                     } else {
-                        rust("Ok(builder.build())")
+                        if (codegenTarget == CodegenTarget.SERVER) {
```

Comment:
having conditional checks littered for client vs server feels not ideal. Is this common in codegen today?

Reply:
Less common than it used to be 😄

Reply:
I think this one can be avoided like we do in `JsonParserGenerator` abstracting with `BuilderInstantiator`.

https://github.com/smithy-lang/smithy-rs/blob/c63e79245430c98b2725627566b80a0fbf8e9619/codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/JsonParserGenerator.kt#L524-L532

Reply:
Good to know. Incorporated in ae02348.